tests/calico_cni_k8s_test.go

```diff
@@ -719,6 +775,19 @@ var _ = Describe("Kubernetes CNI tests", func() {
 							PodCIDR: "10.0.0.0/24",
 						},
 					})
+					if err != nil {
```

Comment:
I'm a bit confused by this as well - why do we need this now but didn't need it before?

Reply:
@lwr20 Could you address this comment?

Reply:
Fine.  Leave it out and see what breaks :)

Reply:
I guess we added `AddNode` in `BeforeEach`, so when creating a node with same name but different PodCIDR, the node could still exists. `DeleteNode` will take some time to remove a node. 

I added function `ensureNodeDeleted` similar to `ensurePodDeleted` and use that to make sure node is not exits before creating a new node.

Reply:
> I guess we added AddNode in BeforeEach

But, we always used to create a node in the BeforeEach AFAICT, we just called it directly. My questions is why do we suddenly need this new logic? It doesn't feel quite right to me but I might just be misunderstanding.

Reply:
We used to create a calico node but not a k8s node I think. 
https://github.com/projectcalico/cni-plugin/blob/master/tests/calico_cni_k8s_test.go#L78

That is why we need a logic now to make sure node is deleted before we create it again.